internal/osutil/file.go

```diff
@@ -36,3 +38,32 @@ func WriteFileWithPermission(path string, data []byte, perm fs.FileMode, overwri
 	}
 	return file.Close()
 }
+
+// CopyToDir copies the src file to dst. Existing file will be overwritten.
+func CopyToDir(src, dst string) (int64, error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return 0, err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return 0, fmt.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return 0, err
+	}
+	defer source.Close()
+
+	if err := os.MkdirAll(dst, 0777); err != nil {
```

Comment:
Should we restrict permission to current user?

Reply:
Note the `os.MkdirAll` FileMode is before unmask. The default Unix permission set for newly created directories is 777 (rwxrwxrwx) masked by the permission bits set in the umask of the process. So I think we should follow the default of Unix here?

Reply:
Do we need to provide read/write permissions to group and public? IMO to start with we can provide read permission but not write permission to group and public.

cc: @shizhMSFT

Reply:
code updated.